src/ain/services/crypto.service.ts

```diff
@@ -36,17 +39,22 @@ export class CryptoService {
       sigBuffer = Buffer.concat([flagByte, sigBuffer]);
       const candidateSig = sigBuffer.toString('base64');
       try {
-        isValid = this.verify(message, address, candidateSig);
+        isValid = this.verify(message, address, candidateSig, Blockchain.DEFICHAIN);
```

Comment:
It's a bit weird to have the blockchain hard coded here. Either the `fallbackVerify` is only used in `verifyDefichain` (which is probably the correct solution) or the blockchain should be added as param.
I think this fallback is used to be able to verify all types of DeFiChain adresses (8xxx, dxxx, df1xxx). Seems like you found a better solution for Bitcoin. Could this also be used for DeFiChain?

Reply:
no, it was just a wrong thought I had yesterday. I have tested BTC addresses (bc1, 3, 1) and all of them worked without fallback. But the fallback should be valid for DEFICHAIN & BITCOIN, only ETHEREUM has nothing to do with that fallback mechanism. Will adjust accordingly